### PR TITLE
[MS Teams] Add initial implementation of the connector

### PR DESCRIPTION
**Pull request by sachinira**
Resolves https://github.com/wso2-enterprise/choreo/issues/5310
Resolves https://github.com/wso2-enterprise/choreo/issues/5311
fixes https://github.com/wso2-enterprise/choreo/issues/6657

**Design document** - https://docs.google.com/document/d/1qKgg4fSlKyOkLR7KDdAjz7xB0aXvYXTIIs0ojRw8fQE/edit?usp=sharing

## Automation tests
 - Unit tests 
   > All the operations are tested separately
 - Integration tests
   > All the operations are tested as a flow

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

https://github.com/sachinira/module-ballerinax-microsoft.teams/tree/teams/teams/samples

 
https://docs.microsoft.com/en-us/graph/api/resources/teams-api-overview?view=graph-rest-1.0
https://docs.microsoft.com/en-us/graph/teams-concept-overview
> MS Teams connector in Logic Apps
https://docs.microsoft.com/en-us/connectors/teams/

### CODE REVIEW

File: teams/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[package]
+org= "ballerinax"
+name= "microsoft.teams"
+version= "0.1.0"
```
Comment (LakshanSS) [inline teams/Ballerina.toml:4]: In the master branch, connector version should be in SNAPSHOT

File: teams/Dependancies.toml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+[[dependency]]
+org = "ballerina"
+name = "url"
+version = "1.1.0-beta.1"
+
+[[dependency]]
+org = "ballerina"
+name = "mime"
+version = "1.1.0-beta.1"
```
Comment (LakshanSS) [inline teams/Dependancies.toml:49]: Since the latest version is beta.2, Change versions to beta.2

File: teams/Dependancies.toml [hunk 3]
```diff
@@ -0,0 +1,49 @@
+[[dependency]]
```
Comment (LakshanSS) [inline teams/Dependancies.toml:1]: Check the name of the file. There is a typo
It should be Dependencies.toml

File: teams/Dependancies.toml (hunk 2 above)
Comment (sachinira) [reply teams/Dependancies.toml]: Changed thanks

File: teams/Ballerina.toml (hunk 1 above)
Comment (sachinira) [reply teams/Ballerina.toml]: Fixed thanks

File: teams/Dependancies.toml (hunk 3 above)
Comment (sachinira) [reply teams/Dependancies.toml]: Fixed this thanks

**Review by abeykoon — APPROVED** [review]

**Pull request merged**